Feature/animal upload

zhihongpang-umbc opened the pull request:
#12 Issue fixed, there is now a working animal upload functionality

vg16855 committed:
Edited models to actually implement a database

fixes #2

SAGE-GARCIA-UMBC committed:
test file

vg16855 committed:
Changed some names and added a __str__ function

vg16855 committed:
Change some names

vg16855 committed:
Merge pull request #3 from ZhihongPang/feature/database-startup

Merge database with develop

SAGE-GARCIA-UMBC committed:
added api paths for players, animals, robots

SAGE-GARCIA-UMBC committed:
test

SAGE-GARCIA-UMBC committed:
added player inventory api, cleaned urlconf files, started beta for battle system

SAGE-GARCIA-UMBC committed:
finished battle demo, finished basic APIs for all 3 models, created a base index html page for the group to work with

SAGE-GARCIA-UMBC committed:
added requirements.txt to standardize python libraries

SAGE-GARCIA-UMBC committed:
resolved merge conf to dev

Commit:
Initial Commit

SAGE-GARCIA-UMBC committed:
Merge pull request #4 from ZhihongPang/feature/battle-system

Merge battle system with develop branch

vg16855 committed:
Updated Database to match ER

I updated the database to match the new ER diagram

Issues:
-The Player's user foreign key and the Animal's animal_class foreign key both require defaults, so I put in null=True to satiate django for now. I will have to make dedicated default functions for the models. I don't know why those two foreign keys are the only one with that requirement. The other foreign keys get along just fine.

LunaXIII committed:
Added Background image and changed front page image from tigers.jpg to pika.jpg.

LunaXIII committed:
Create.gitignore

LunaXIII committed:
Create.gitignore

LunaXIII committed:
Fixed Border issue.

vg16855 committed:
Commit on 4-13-2022 at 7:06 PM

Updated readme to explain some stuff

zhihongpang-umbc committed:
Merge pull request #6 from ZhihongPang/feature/database-startup

Feature/database startup

LunaXIII committed:
Updated Models.py

vg16855 committed:
Commit on 4-13-2022 at 11:35 PM

Fixed attributes that used the "choice" field using the wrong syntax

vg16855 committed:
Commit on 4-13-2022

Actually migrated the database

vg16855 committed:
Merge branch 'feature/database-startup' of https://github.com/ZhihongPang/Pokemon530 into feature/database-startup

zhihongpang-umbc committed:
Merge pull request #7 from ZhihongPang/feature/database-startup

Feature/database startup pull request

LunaXIII committed:
Update Models.py

vg16855 committed:
Merge pull request #5 from ZhihongPang/feature/login_page

Initial Commit

zhihongpang-umbc committed:
Animal Upload

A basic animal upload, accessible via localhost/animals

zhihongpang-umbc committed:
Merge branch 'develop' into feature/animal-upload

vg16855 commented:
Code Review: This branch implements the actual uploading of photos and adding of new animals to the database. This adds the ability to view the working directory. There is also a new URL that links to a local directory that stores images(of the animals). Right now, this branch has been tested with a smaller version of the database,(namely because we don't have features for adding players yet). So we will hold off on merging for now. This seems to be a story that will span multiple sprints.

zhihongpang-umbc committed:
Added tests for animal upload

Test cases for animal upload added, the tests mainly test to see if the Animals initialized are updated correctly in the database.

vg16855 commented:
I think I'll merge it, since all we need to do is uncomment the removed fields.

SAGE-GARCIA-UMBC commented:
This was merged from feature to main branch. Was this intentional? @ZhihongPang @vg16855
